plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php

```diff
@@ -52,17 +52,27 @@ public function handle_template_include( $template ) {
 
 		$coming_soon_template = get_query_template( 'coming-soon' );
 
-		if ( ! wc_current_theme_is_fse_theme() && $this->coming_soon_helper->is_store_coming_soon() ) {
+		$is_fse_theme         = wc_current_theme_is_fse_theme();
+		$is_store_coming_soon = $this->coming_soon_helper->is_store_coming_soon();
+
+		if ( ! $is_fse_theme && $is_store_coming_soon ) {
 			get_header();
 		}
 
+		add_action(
+			'wp_head',
+			function () {
+				echo "<meta name='woo-coming-soon-page' content='yes'>";
+			}
+		);
```

Comment:
We can also consider adding a custom HTTP header. It's useful for server side checks so we don't need to use `strpos` to check if the page is a coming soon page. 


```php
add_action('send_headers', function add_custom_http_header() {
    header("X-Woo-Coming-Soon-Page: true");
  }
);
```

Reply:
@chihsuan That's a great idea! cc @rjchow